### DISCUSSION TITLE
Which is the fastest way to bootstrap a local Milvus for integrations tests

### DISCUSSION THREAD

**Discussion by eolivelli** [milvus-io/milvus#27358] [Ideas & Feature requests]
Hello,
I am one of the maintainers of [LangStream](https://github.com/LangStream/langstream).

I have been working on an integration with Milvus, and everything is working well.

But I am not able to find a way to bootstrap Milvus in docker with a limited set of resources and a smell startup time.
Especially on M1 laptops.

So my question is: which is the best way to bootstrap Milvus for unit tests ? I am on Java.

We are also discussion about using Milvus.io as default Vector database for the "getting started" guides, but starting Milvus lite still takes minutes on my (M1) laptop from the terminal and it times out starting in a docker container.

Looking forward for guidance

**Comment by filip-halt** [comment]
Embedded Milvus is the most lightweight version that we have. If that is not good enough you might be able to grab a Milvus image and try to import the tests into the image and run it there, don't think it will be faster. In most cases for third party dependencies like these its best to just mock the Milvus functionality and rely on the projects tests to cover whats needed.

**Comment by xiaofan-luan** [comment]
Easiest way might be using milvus-lite, see https://github.com/milvus-io/milvus-lite
